dace/data.py

```diff
@@ -57,9 +57,9 @@ class Data(object):
                        default=dace.dtypes.StorageType.Default,
                        from_string=lambda x: dtypes.StorageType[x])
     location = Property(
-        dtype=str,  # Dict[str, symbolic]
```

Comment:
we have the class `DictProperty` especially designed for this, with serialization support.

Reply:
How does this work for default? We need to set this now because of serialization, right?

Reply:
`DictProperty` has `value_type` as mandatory argument, but location does not enforce this. Should I change `DictProperty` so this is optional?